cli/flox/src/commands/list.rs

```diff
@@ -255,40 +254,40 @@ impl List {
         Ok(())
     }
 
-    /// Read existing lockfile or resolve to create a new [LockedManifest].
+    /// Read existing lockfile or lock to create a new [LockedManifest].
     ///
-    /// Does not write the lockfile,
-    /// as that would require writing to the environment in case of remote environments)
+    /// This may write the lockfile depending on the type of environment;
+    /// path and managed environments with local checkouts will write the
+    /// lockfile.
     fn get_lockfile(flox: &Flox, env: &mut dyn Environment) -> Result<LockedManifest> {
-        let lockfile_path = env
-            .lockfile_path(flox)
-            .context("Could not get lockfile path")?;
-
-        let lockfile = if !lockfile_path.exists() {
-            debug!("No lockfile found, locking environment...");
-            Dialog {
+        let lockfile = Dialog {
                 message: "No lockfile found for environment, building...",
                 help_message: None,
-                typed: Spinner::new(|| env.lock(flox)),
+                typed: Spinner::new(|| env.deserialized_lockfile(flox)),
             }
-            .spin()?
-        } else {
-            debug!("Using existing lockfile");
-            // we have already checked that the lockfile exists
-            let path = CanonicalPath::new(lockfile_path).unwrap();
-            LockedManifest::read_from_file(&path)?
-        };
-
+            // TODO: it would be better if we knew when a lock was actually happening
+            .spin_with_delay(Duration::from_secs_f32(0.25))?;
         Ok(lockfile)
     }
 }
```

Comment:
question: does the "relock" if the lockfile is out of data now?

Reply:
Path and managed environments will relock if the lockfile is out of date, remote environments won't ever relock

Reply:
Tweaked the comment